### DISCUSSION TITLE
Slow performance for large InputTypes

### DISCUSSION THREAD

**Discussion by run3wide** [MichalLytek/type-graphql#1402] [General]
Our team experiences large latencies when inputs become large, specifically large arrays. I've profiled our app and tracked down most of the time is spent within `function convertToInput(tree, data)`. I was able to increase the performance by using a GraphQLJsonObject, but I lose documentation and validation with that approach. Do you have any recommendations for increasing performance?

**Comment by MichalLytek** [comment]
All we have for performance tweaks is https://typegraphql.com/docs/performance.html
